File: autoscale/src/main/java/com/sequenceiq/periscope/repository/HistoryRepository.java
package com.sequenceiq.periscope.repository;

import java.util.List;

import javax.transaction.Transactional;

import org.springframework.data.domain.Pageable;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.Param;

import com.sequenceiq.cloudbreak.workspace.repository.EntityType;
import com.sequenceiq.periscope.domain.History;

@EntityType(entityClass = History.class)
@Transactional(Transactional.TxType.REQUIRED)
public interface HistoryRepository extends CrudRepository<History, Long> {

    List<History> findByClusterId(@Param("clusterId")Long clusterId, Pageable pageRequest);
}

File: autoscale/src/main/java/com/sequenceiq/periscope/repository/HistoryRepository.kt
package com.sequenceiq.periscope.repository

import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.query.Param
import org.springframework.security.access.prepost.PostAuthorize

import com.sequenceiq.periscope.domain.History

interface HistoryRepository : CrudRepository<History, Long> {

    fun findAllByCluster(@Param("id") id: Long?): List<History>

    @PostAuthorize("hasPermission(returnObject,'read')")
    fun findByCluster(@Param("clusterId") clusterId: Long?, @Param("historyId") historyId: Long?): History

}
